src/i2c/src/i2c.rs

```diff
@@ -313,16 +321,10 @@ impl<D: I2cDevice> I2cAdapter<D> {
     // Creates a new adapter corresponding to the specified number.
     fn new(adapter_no: u32) -> Result<I2cAdapter<D>> {
         let i2cdev = format!("/dev/i2c-{}", adapter_no);
-        let func: u64 = I2C_FUNC_SMBUS_ALL;
         let mut device = D::open(i2cdev)?;
+        let func = device.funcs()?;
```

Comment:
I just realized that testing that this bug does not resurface is not super straightforward with the current abstractions that we have in place. We will need to work more on the higher level abstractions so that we can pass a device that is initialized by us in the tests. I can provide a prototype for this a bit later in the week. For now we can merge this without a test.